addons/mrp_operations/mrp_operations.py

```diff
@@ -295,18 +295,21 @@ def _move_pass(self, cr, uid, ids, context=None):
                 if l.state in ('done','cancel','draft'):
                     continue
                 todo += l.move_dest_id_lines
-                if l.production_id and (l.production_id.date_finished > dt):
-                    if l.production_id.state not in ('done','cancel'):
-                        for wc in l.production_id.workcenter_lines:
-                            i = self.pool.get('resource.calendar').interval_min_get(
-                                cr,
-                                uid,
-                                wc.workcenter_id.calendar_id.id or False,
-                                dt, wc.hour or 0.0
-                            )
-                            dt = i[0][0]
-                        if l.production_id.date_start > dt.strftime('%Y-%m-%d %H:%M:%S'):
-                            self.write(cr, uid, [l.production_id.id], {'date_start':dt.strftime('%Y-%m-%d %H:%M:%S')}, mini=True)
+                if l.production_id and l.production_id.date_finished:
+                    date_end = l.production_id.date_finished
+                    d_end = datetime.strptime(date_end, '%Y-%m-%d %H:%M:%S')
+                    if d_end > dt:
```

Comment:
please avoid changing the indentation by adding an if. Something like:

```
date_end = l.production_id.date_finished
if date_end and datetime.strptime(date_end, '%Y-%m-%d %H:%M:%S') > dt:
```
The syntax `l.production_id and l.production_id.date_finished` is no longer needed in 8.0, `date_finished` will be `False` if `production_id` is not set.